### ISSUE TITLE
DomSanitizer does not protect against target=_blank exploit

### ISSUE THREAD

**Issue by aeslinger0** [angular/angular#33083]
**TLDR**: DomSanitizer should add `rel='noopener noreferrer'` to anchor tags that target a new window.

I just read [this article](https://www.pixelstech.net/article/1537002042-The-danger-of-target%3D_blank-and-opener) which says that anchor tags that target a new window are susceptible to phishing attacks.

Unfortunately, the DomSanitizer does not seem to handle the exploit mentioned in the article. I recommend that you add `rel='noopener noreferrer'` (where it does not already exist) to any hyperlink that has a target other than _self. The article only mentions "_blank", but I would think it would be a problem for any link that opens in a new window, such as target="myNamedTarget".

**Comment by Airblader** [comment]
There are absolutely valid cases *not* to add this. It's only important for link targets you do not control (either link itself or the targeted website), but apps may well open other parts of the app (or other apps managed by them) and keeping opener / referrer might be important there.

**Comment by Airblader** [comment]
This isn't really an issue of sanitization anyway, I would move something like this into a linting rule.

**Comment by aeslinger0** [comment]
@Airblader, what about the case when someone allow users to post HTML content to their page and runs the content through DomSanitizer first? Linting won't fix that. Maybe the DomSanitizer should be configurable for each particular use case.

**Comment by Airblader** [comment]
I haven't said DomSanitizer should be replaced by linting, I said that noopener and noreferrer on _blank links should be handled with linting.

**Comment by Airblader** [comment]
That is a good point, yes. I agree that this is something to consider including indeed.

**Comment by JeanMeche** [comment]
Yes we can close this, `_blank` implies `noopener` on all supported browsers 👍  
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#browser_compatibility
